### PR TITLE
Pin compatible numpy for older HTSEQ versions

### PR DESCRIPTION
**Pull request by lparsons**
This is required to keep older versions of HTSEQ functional

* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [X] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by lparsons** [comment]
This appears to be working, but I have one question: Is there some way to test that this fixes the issue of HTSeq getting installed with an incompatible numpy before merging? I have been unable to reproduce the error in testing since the tests rebuild the packages with the new numpy.

**Comment by bgruening** [comment]
> This appears to be working, but I have one question: Is there some way to test that this fixes the issue of HTSeq getting installed with an incompatible numpy before merging? I have been unable to reproduce the error in testing since the tests rebuild the packages with the new numpy.

You mean htseq as a dependency of an other package? You could include in this PR to recipes one depends on the other, this should work.

File: recipes/htseq/0.6.1/meta.yaml [hunk 1]
```diff
@@ -12,6 +12,8 @@ package:
   version: 0.6.1
 
 requirements:
+  build:
```
Comment (bgruening) [inline recipes/htseq/0.6.1/meta.yaml]: Build number increase in this recipe please.

Comment (lparsons) [reply recipes/htseq/0.6.1/meta.yaml]: No changes to that recipe.

Comment (lparsons) [reply recipes/htseq/0.6.1/meta.yaml]: Oh, wait, I see there is one... Sorry. I'm actually not clear if that's needed. I put that in attempting to fix `circleci build` errors, but I never got that working.

**Comment by lparsons** [comment]
@bgruening I don't see how this tests anything. If I understand the problem correctly, the issue is that HTSeq was built with an older numpy, but when it's installed now, a newer, incompatible numpy is installed with it. This update to the recipe triggers a new build, which is with the new numpy, and thus works. The question is, will it break in the future? I guess no way to know.

File: recipes/htseq/0.6.1/meta.yaml [hunk 2]
```diff
@@ -12,6 +12,8 @@ package:
...
+  - {{ compiler('c') }}
```
Comment (lparsons) [inline recipes/htseq/0.6.1/meta.yaml]: Is this required/recommended? I added it in an attempt to get `circleci build` tests to pass, but they never did. Not sure if I should keep it or not.

Comment (bgruening) [reply recipes/htseq/0.6.1/meta.yaml]: yes it is, if a compiler is needed :)

**Comment by bgruening** [comment]
Thanks!

**Pull request merged**